suite-native/biometrics/src/useBiometrics.tsx

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useState } from 'react';
+import { AppState, Platform } from 'react-native';
+
+import * as LocalAuthentication from 'expo-local-authentication';
+
+import { useAlert } from '@suite-native/alerts';
+
+import { useIsBiometricsEnabled } from './biometricsAtoms';
+import { getIsBiometricsFeatureAvailable } from './isBiometricsFeatureAvailable';
+
+export const useBiometrics = () => {
+    const { showAlert, hideAlert } = useAlert();
+    const { isBiometricsOptionEnabled, setIsBiometricsOptionEnabled } = useIsBiometricsEnabled();
+    const [isUserAuthenticated, setIsUserAuthenticated] = useState(false);
+
+    const authenticate = useCallback(async () => {
+        if (Platform.OS === 'android') {
+            // Cancel any previous authentication attempt. (Only available for android)
+            await LocalAuthentication.cancelAuthenticate();
+        }
+
+        if (AppState.currentState !== 'active') return;
+
+        try {
+            const result = await LocalAuthentication.authenticateAsync();
+            console.log('result', result);
+
+            if (
+                !result.success &&
+                (result.error === 'user_cancel' ||
+                    result.error === 'system_cancel' ||
+                    result.error.startsWith('unknown'))
+            ) {
+                showAlert({
+                    title: 'Authentication canceled',
+                    description: 'You will have to try again.',
+                    pictogramVariant: 'red',
+                    icon: 'warningCircle',
+                    onPressPrimaryButton: authenticate,
+                    primaryButtonTitle: 'Try again',
+                    secondaryButtonTitle: 'Cancel',
+                    onPressSecondaryButton: () => null,
+                });
+            } else {
+                setIsUserAuthenticated(result.success);
+                hideAlert();
```

Comment:
Yes, but what happened to me during testing sometimes was that we displayed alert at the same time of user trying to authenticate. I wasn't able to reproduce it and since it's native code I'm not able to determine what was causing the issue. I will add a comment here so it's clear. https://github.com/trezor/trezor-suite/pull/8610/commits/6024ce3f54c0972cf04102f9de263d412785905d